File: .skills/openclaw-skills/skills/binkosun/img2img/gen_image.js
#!/usr/bin/env node
/** 图生图 - 使用 DALL-E 3 生成图片 */

const { OpenAI } = require('openai');
const https = require('https');

async function generateImage(prompt) {
  const apiKey = process.env.API_KEY;
  
  if (!apiKey) {
    console.error('错误: 未设置 API_KEY 环境变量');
    process.exit(1);
  }

  const client = new OpenAI({
    apiKey: apiKey,
    baseURL: process.env.BASE_URL || process.env.API_BASE_URL || 'https://api.openai.com/v1',
    httpAgent: new https.Agent({ rejectUnauthorized: false })
  });

  const response = await client.images.generate({
    model: 'dall-e-3',
    prompt: prompt,
    size: '1024x1024',
    quality: 'standard',
    n: 1,
  });

  console.log(response.data[0].url);
}

const prompt = process.argv.slice(2).join(' ');
if (!prompt) {
  console.error('用法: node gen_image.js "图片描述"');
  process.exit(1);
}

generateImage(prompt).catch(err => {
  console.error('错误:', err.message);
  process.exit(1);
});

File: .skills/openclaw-skills/skills/binkosun/img2img/gen_image.py
#!/usr/bin/env python3
"""图生图 - 使用 DALL-E 3 生成图片"""

import os
import sys
import json
import argparse

def generate_image(prompt: str, api_key: str = None) -> str:
    """调用 DALL-E 3 API 生成图片"""
    import openai
    
    if not api_key:
        api_key = os.environ.get("API_KEY")
    
    if not api_key:
        print("错误: 未设置 API_KEY 环境变量", file=sys.stderr)
        sys.exit(1)
    
    client = openai.OpenAI(api_key=api_key)
    
    response = client.images.generate(
        model="dall-e-3",
        prompt=prompt,
        size="1024x1024",
        quality="standard",
        n=1,
    )
    
    return response.data[0].url


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="DALL-E 图生图")
    parser.add_argument("prompt", help="图片描述")
    parser.add_argument("--api-key", help="OpenAI API Key")
    
    args = parser.parse_args()
    
    try:
        url = generate_image(args.prompt, args.api_key)
        print(url)
    except Exception as e:
        print(f"错误: {e}", file=sys.stderr)
        sys.exit(1)
